### PR TITLE
SREP-2178 Add dedicated-admins permissions to customer monitoring NS

### PR DESCRIPTION
**Pull request by rogbas**
- Incorporate RBAC permissions for dedicated-admins in `openshift-customer-monitoring` NS
     - Only a pre-selected list of secrets can be managed by dedicated-admins in that NS
- Add 3 new secrets to the list owned by dedicated-admins in the same NS
     - prometheus-additional-scrape-config
     - prometheus-additional-alertmanager-config
     - alertmanager-config

cc @aditya-konarde

### CODE REVIEW

File: deploy/osd-customer-monitoring/05-role.yaml [hunk 1]
```diff
@@ -39,4 +42,491 @@ rules:
...
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - ""
+  - apps.openshift.io
+  resources:
+  - deploymentconfigrollbacks
```
Comment (jharrington22) [inline deploy/osd-customer-monitoring/05-role.yaml]: Is this right?

File: deploy/osd-customer-monitoring/05-role.yaml [hunk 2]
```diff
@@ -39,4 +42,491 @@ rules:
...
+  verbs:
+  - create
+- apiGroups:
+  - ""
+  - project.openshift.io
+  resources:
+  - projects
+  verbs:
+  - get
+- apiGroups:
```
Comment (jharrington22) [inline deploy/osd-customer-monitoring/05-role.yaml]: Out of order?

File: deploy/osd-customer-monitoring/05-role.yaml [hunk 3]
```diff
@@ -39,4 +42,491 @@ rules:
   - servicemonitors
   - podmonitors
   verbs:
-  - "*"
\ No newline at end of file
+  - "*"
+- apiGroups:
+  - ""
```
Comment (jharrington22) [inline deploy/osd-customer-monitoring/05-role.yaml]: Create secrets in any namespace?

File: deploy/osd-customer-monitoring/05-role.yaml [hunk 4]
```diff
@@ -39,4 +42,491 @@ rules:
...
+  - authorization.openshift.io
+  resources:
+  - resourceaccessreviews
+  - subjectaccessreviews
+  verbs:
+  - create
+- apiGroups:
+  - ""
```
Comment (jharrington22) [inline deploy/osd-customer-monitoring/05-role.yaml]: Is this required? There are a few apiGroups like this.

File: deploy/osd-customer-monitoring/05-role.yaml (hunk 1 above)
Comment (rogbas) [reply deploy/osd-customer-monitoring/05-role.yaml]: It's part of the dedicated-admin set of permissions that we support today.

File: deploy/osd-customer-monitoring/05-role.yaml (hunk 2 above)
Comment (rogbas) [reply deploy/osd-customer-monitoring/05-role.yaml]: I will remove lines 59 and 60

File: deploy/osd-customer-monitoring/05-role.yaml (hunk 4 above)
Comment (rogbas) [reply deploy/osd-customer-monitoring/05-role.yaml]: I'm following the same set of permissions we have for dedicated-admins today, in the original Role(s) there is this "" apiGroup as well. I'm assuming it's there for a reason, probably retro compatibility?

Comment (jharrington22) [reply deploy/osd-customer-monitoring/05-role.yaml]: I would have thought that `""` gives access to all apiGroups? or is that specifically `"*"`?

Comment (cblecker) [reply deploy/osd-customer-monitoring/05-role.yaml]: it's a blank API group. It's not really needed in either case.

The only resources that need it are:
```
$ oc api-resources --api-group=""
NAME                     SHORTNAMES   APIGROUP   NAMESPACED   KIND
bindings                                         true         Binding
componentstatuses        cs                      false        ComponentStatus
configmaps               cm                      true         ConfigMap
endpoints                ep                      true         Endpoints
events                   ev                      true         Event
limitranges              limits                  true         LimitRange
namespaces               ns                      false        Namespace
nodes                    no                      false        Node
persistentvolumeclaims   pvc                     true         PersistentVolumeClaim
persistentvolumes        pv                      false        PersistentVolume
pods                     po                      true         Pod
podtemplates                                     true         PodTemplate
replicationcontrollers   rc                      true         ReplicationController
resourcequotas           quota                   true         ResourceQuota
secrets                                          true         Secret
serviceaccounts          sa                      true         ServiceAccount
services                 svc                     true         Service
```

File: deploy/osd-customer-monitoring/05-role.yaml (hunk 2 above)
Comment (rogbas) [reply deploy/osd-customer-monitoring/05-role.yaml]: removed

File: deploy/osd-customer-monitoring/05-role.yaml (hunk 4 above)
Comment (rogbas) [reply deploy/osd-customer-monitoring/05-role.yaml]: +1, "" is not "*". It's just a set of resources that can have apiGroup = "", examples: configmaps, endpoints, services, secrets.

Thanks for the list @cblecker

Comment (rogbas) [reply deploy/osd-customer-monitoring/05-role.yaml]: To clean it up, I removed all the "" apiGroups that already had another apiGroup assigned in the rule

Comment (jharrington22) [reply deploy/osd-customer-monitoring/05-role.yaml]: Interesting thanks @cblecker for the explanation

**Pull request merged**